File: python/PowerSupply.py
from Komponen import Komponen

class PowerSupply(Komponen):
    def __init__(self, id='', warranty_months=0, merk='', nama='', harga=0.0,
                 wattage=0, certification='', modular=False):
        super().__init__(id, warranty_months, merk, nama, harga)
        self.wattage = wattage
        self.certification = certification
        self.modular = modular

    def set_wattage(self, wattage):
        self.wattage = wattage

    def set_certification(self, certification):
        self.certification = certification

    def set_modular(self, modular):
        self.modular = modular

    def get_wattage(self):
        return self.wattage

    def get_certification(self):
        return self.certification

    def is_modular(self):
        return self.modular

    def display_info(self):
        super().display_info()
        print(f"PSU Specs: {self.wattage}W, {self.certification} certified")
        print(f"Modular: {'Yes' if self.modular else 'No'}")

File: java/PowerSupply.java
public class PowerSupply extends Komponen {
    private int wattage;
    private String certification;
    private boolean modular;

    public PowerSupply() {
        super();
        this.wattage = 0;
        this.certification = "";
        this.modular = false;
    }

    public PowerSupply(String id, int warrantyMonths, String merk, String nama, double harga,
                       int wattage, String certification, boolean modular) {
        super(id, warrantyMonths, merk, nama, harga);
        this.wattage = wattage;
        this.certification = certification;
        this.modular = modular;
    }

    public void setWattage(int wattage) {
        this.wattage = wattage;
    }

    public void setCertification(String certification) {
        this.certification = certification;
    }

    public void setModular(boolean modular) {
        this.modular = modular;
    }

    public int getWattage() {
        return wattage;
    }

    public String getCertification() {
        return certification;
    }

    public boolean isModular() {
        return modular;
    }

    @Override
    public void displayInfo() {
        super.displayInfo();
        System.out.println("PSU Specs: " + wattage + "W, " + certification + " certified");
        System.out.println("Modular: " + (modular ? "Yes" : "No"));
    }
}
